tests/post/steps/test_authentication.py

```diff
@@ -271,4 +273,28 @@ def _dex_auth_request(control_plane_ip, username, password):
     return result
 
 
+# a retry helper that performs 3 retries with an exponential sleep interval
+# between each request.
+# Only retry internal server errors and service unavailable errors
+def requests_retry_session(
+    retries=3,
+    backoff_factor=0.3,
+    status_forcelist=(500, 503),
+    method_whitelist=frozenset(['GET', 'POST']),
+    session=None
+):
+    session = session or requests.Session()
+    retry = Retry(
+        total=retries,
+        read=retries,
+        connect=retries,
+        backoff_factor=backoff_factor,
+        status_forcelist=status_forcelist,
+    )
+    adapter = HTTPAdapter(max_retries=retry)
+    session.mount("http://", adapter)
+    session.mount("https://", adapter)
+    return session
+
+
```

Comment:
Straight from the blog post, the nice thing to do is mention it in a comment (https://www.peterbe.com/plog/best-practice-with-retries-with-requests).

Reply:
Added as comment before the helper definition